### PR TITLE
Bug fix for notifications API

### PR DESCRIPTION
**Pull request by lumberj**

### CODE REVIEW

**Comment by wltsmrz** [comment]
LGTM

File: lib/utils.js [hunk 1]
```diff
@@ -91,3 +92,22 @@ function parseCurrencyQuery(query) {
...
+ *  Order two rippled transactions based on their ledger_index.
+ *  If two transactions took place in the same ledger, sort
+ *  them based on TransactionIndex
+ *  See: https://ripple.com/build/transactions/
+ *
+ *  @param {Object} first
+ *  @param {Object} second
+ *  @returns {Number} [-1, 1]
+ */
+function compareTransactions(first, second) {
```
Comment (clark800) [inline lib/utils.js]: For the sake of being standard, should return 0 if the two are equal.

File: test/unit/utils-test.js [hunk 2]
```diff
@@ -173,3 +173,36 @@ suite('unit - utils.parseCurrencyQuery()', function() {
     });
   });
 });
+
+
+suite('unit - utils.compareTransactions()', function() {
+  test('compareTransactions() -- different ledgers', function() {
+    var tx1 = {
+      ledger_index: '1'
```
Comment (clark800) [inline test/unit/utils-test.js]: Is this supposed to be a string?

File: test/unit/utils-test.js [hunk 3]
```diff
@@ -173,3 +173,36 @@ suite('unit - utils.parseCurrencyQuery()', function() {
...
+    };
+
+    var tx2 = {
+      ledger_index: '2'
+    };
+
+    assert.deepEqual(utils.compareTransactions(tx1,tx2), -1);
```
Comment (clark800) [inline test/unit/utils-test.js]: Since this is comparing numbers, maybe use strictEqual

File: lib/utils.js (hunk 1 above)
Comment (lumberj) [inline lib/utils.js]: Done

File: test/unit/utils-test.js (hunk 3 above)
Comment (lumberj) [inline test/unit/utils-test.js]: Changing

File: lib/utils.js [hunk 4]
```diff
@@ -91,3 +92,24 @@ function parseCurrencyQuery(query) {
...
+ *
+ *  @param {Object} first
+ *  @param {Object} second
+ *  @returns {Number} [-1, 0, 1]
+ */
+function compareTransactions(first, second) {
+  if (first.ledger_index === second.ledger_index) {
+    const order = first.meta.TransactionIndex - second.meta.TransactionIndex;
+
+    return (order === 0) ? 0 : (order > 0 ? 1 : -1);
```
Comment (clark800) [inline lib/utils.js]: Extract method "signum"

**Comment by geertweening** [comment]
LGTM

File: lib/utils.js (hunk 4 above)
Comment (lumberj) [inline lib/utils.js]: Extracted

**Comment by clark800** [comment]
LGTM

**Pull request merged**